src/huggingface_hub/file_download.py

```diff
@@ -1394,6 +1437,13 @@ def _resumable_file_manager() -> Generator[io.BufferedWriter, None, None]:
         with temp_file_manager() as temp_file:
             logger.info("downloading %s to %s", url, temp_file.name)
 
+            if local_dir is None:
+                target_dir = os.path.dirname(blob_path)
+            else:
+                target_dir = os.path.dirname(local_dir)
+
+            _check_disk_space(expected_size, temp_file, target_dir)
```

Comment:
With the update that I suggested above, the logic here change slightly to something like this:

```py
            if expected_size is not None: # might be None if HTTP header not set correctly
                # Check tmp path
                _check_disk_space(expected_size, os.path.dirname(temp_file.name))

                # Check destination
                _check_disk_space(expected_size, os.path.dirname(blob_path))
                if local_dir is not None:
                    _check_disk_space(expected_size, local_dir)
```